pkg/ingress/model_build_load_balancer_test.go

```diff
@@ -682,6 +683,924 @@ func Test_defaultModelBuildTask_buildLoadBalancerName(t *testing.T) {
+		VpcId: awssdk.String("vpc-1"),
+	}
+	sg14 = &ec2.SecurityGroup{
+		GroupId: awssdk.String("sg-14"),
+		Tags: []*ec2.Tag{
+			{
+				Key:   awssdk.String("tagtest2"),
+				Value: awssdk.String("1"),
+			},
+		},
+		VpcId: awssdk.String("vpc-1"),
+	}
+	sg15 = &ec2.SecurityGroup{
+		GroupId: awssdk.String("sg-15"),
+		Tags: []*ec2.Tag{
+			{
+				Key:   awssdk.String("tagtest2"),
+				Value: awssdk.String("1"),
+			},
+			{
+				Key:   awssdk.String("kubernetes.io/cluster/test-cluster"),
+				Value: awssdk.String("shared"),
+			},
+		},
+		VpcId: awssdk.String("vpc-1"),
+	}
+)
+
+func stubDescribeSecurityGroupsAsList(ctx context.Context, input *ec2.DescribeSecurityGroupsInput) ([]*ec2.SecurityGroup, error) {
+	sgs := []*ec2.SecurityGroup{
+		sg1,
+		sg2,
+		sg3,
+		sg4,
+		sg5,
+		sg6,
+		sg7,
+		sg8,
+		sg9,
+		sg10,
+		sg11,
+		sg12,
+		sg13,
+		sg14,
+		sg15,
+	}
+	if input.GroupIds != nil {
+		var filtered []*ec2.SecurityGroup
+		for _, sg := range sgs {
+			for _, id := range input.GroupIds {
+				if awssdk.StringValue(sg.GroupId) == awssdk.StringValue(id) {
+					filtered = append(filtered, sg)
+					continue
+				}
+			}
+		}
+		sgs = filtered
+	}
+	if input.Filters != nil {
+		var filtered []*ec2.SecurityGroup
+	sgLoop:
+		for _, sg := range sgs {
+			for _, filter := range input.Filters {
+				eligible := false
+				if awssdk.StringValue(filter.Name) == "vpc-id" {
+					for _, name := range filter.Values {
+						if awssdk.StringValue(sg.VpcId) == awssdk.StringValue(name) {
+							eligible = true
+							continue
+						}
+					}
+				} else if strings.HasPrefix(awssdk.StringValue(filter.Name), "tag:") {
+					key := strings.TrimPrefix(awssdk.StringValue(filter.Name), "tag:")
+					for _, value := range filter.Values {
+						for _, tag := range sg.Tags {
+							if awssdk.StringValue(tag.Key) == key && awssdk.StringValue(tag.Value) == awssdk.StringValue(value) {
+								eligible = true
+								continue
+							}
+						}
+					}
+				} else {
+					return nil, fmt.Errorf("unexpected filter %q", awssdk.StringValue(filter.Name))
+				}
+				if !eligible {
+					continue sgLoop
+				}
+			}
+			filtered = append(filtered, sg)
+		}
+		sgs = filtered
+	}
+	return sgs, nil
+}
+
+func Test_defaultModelBuildTask_buildLoadBalancerSecurityGroups(t *testing.T) {
+	type fields struct {
+		ingGroup        Group
+		scheme          elbv2.LoadBalancerScheme
+		enableBackendSG bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []string
+		wantErr string
+	}{
+		{
+			name: "no annotation managed SG",
+			fields: fields{
+				ingGroup: Group{
+					ID: GroupID{Namespace: "awesome-ns", Name: "ing-1"},
+					Members: []ClassifiedIngress{
+						{
+							Ing: &networking.Ingress{
+								ObjectMeta: metav1.ObjectMeta{
+									Namespace: "awesome-ns",
+									Name:      "ing-1",
+								},
+							},
+						},
+					},
+				},
+				scheme: elbv2.LoadBalancerSchemeInternetFacing,
+			},
+			want: []string{"sg-managed"},
```

Comment:
Should the default value of `--enable-backend-security-group` is true? The controller should always create both frontend and backend SGs to ALB if there's no annotation or spec.SecurityGroups specified?